javascript/progressbar.js

```diff
@@ -79,11 +79,12 @@ function requestProgress(id_task, progressbarContainer, gallery, atEnd, onProgre
     var wakeLock = null;
 
     var requestWakeLock = async function() {
-        if (!opts.prevent_screen_sleep_during_generation || wakeLock) return;
+        if (!opts.prevent_screen_sleep_during_generation || wakeLock !== null) return;
         try {
             wakeLock = await navigator.wakeLock.request('screen');
         } catch (err) {
             console.error('Wake Lock is not supported.');
+            wakeLock = false;
```

Comment:
base on example on https://developer.mozilla.org/en-US/docs/Web/API/WakeLock/request#examples
> The wake lock request fails - usually system-related, such as low battery.

my understanding is that it can fail due to other reasons
in cases wake lock is temporarily unavailable (such as low battery) you will want to re-request the lock to see if it's avable

your PR would prevent any future attempts of requesting wake lock if it fails due to any other reasons just one

if this is going to be merged it should be able to distinguish between error resons

Reply:
實際上不太需要擔心失敗後 以後就不會在嘗試 Wake Lock 的問題
因為每次請求時都會重設
沒必要在同一次請求內 不斷的去試著 Wake Lock 直到成功或請求結束

https://github.com/search?q=repo%3AAUTOMATIC1111%2Fstable-diffusion-webui%20requestProgress&type=code

Reply:
oh yeah you're right
I thought `var wakeLock` was global